components/renku_data_services/project/api.spec.yaml

```diff
@@ -0,0 +1,382 @@
+            schema:
+              $ref: "#/components/schemas/ProjectPut"
+      responses:
+        "200":
+          description: The project
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Project"
+        "404":
+          description: The project does not exist
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+    patch:
+      summary: Update specific fields of an existing project
+      parameters:
+        - in: path
+          name: project_id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ProjectPatch"
+      responses:
+        "200":
+          description: The project
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Project"
+        "404":
+          description: The project does not exist
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+    delete:
+      summary: Remove a project
+      parameters:
+        - in: path
+          name: project_id
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: The project was removed or did not exist in the first place
+        default:
+          $ref: '#/components/responses/Error'
+      tags:
+        - projects
+  /version:
+    get:
+      summary: Get the version of the service
+      responses:
+        "200":
+          description: The error
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Version"
+
+components:
+  schemas:
+    Project:
+      type: object
+      description: A Renku native project definition and metadata
+      additionalProperties: false
+      properties:
+        id:
+          $ref: "#/components/schemas/Ulid"
+        name:
+          $ref: "#/components/schemas/Name"
+        slug:
+          $ref: "#/components/schemas/Slug"
+        creation_date:
+          $ref: "#/components/schemas/CreationDate"
+        created_by:
+          $ref: "#/components/schemas/User"
+        repositories:
+          $ref: "#/components/schemas/RepositoriesList"
+        visibility:
+          $ref: "#/components/schemas/Visibility"
+        members:
+          $ref: "#/components/schemas/MembersWithRoles"
+      required:
+        - "id"
+        - "name"
+        - "slug"
+        - "created_by"
+        - "visibility"
```

Comment:
These two should not be requried. Visibility can default to `private`.

For created by it should just not be an option at all to change. You will fill this in from the access token of the request. This way I can create a project logged in as myself but put Sekhar as the creator.

Reply:
I agree with the above and also think that "id" should not be a parameter -- the server should generate it.

Reply:
This is `Project` schema which is always used in responses. These fields should be required for a Renku project.

I believe your comments refers to `ProjectPatch` and `ProjectPut` schemas that mark some of these fields as required. I fixed those two.